lib/CalEventParser.ts

```diff
@@ -1,151 +1,14 @@
-<strong>${this.calEvent.language("invitee_email")}:</strong><br /><a href="mailto:${
-        this.calEvent.attendees[0].email
-      }">${this.calEvent.attendees[0].email}</a><br />
-` +
-      (this.getLocation()
-        ? `<strong>${this.calEvent.language("location")}:</strong><br />${this.getLocation()}<br />
-`
-        : "") +
-      `<strong>${this.calEvent.language("invitee_timezone")}:</strong><br />${
-        this.calEvent.attendees[0].timeZone
-      }<br />
-<strong>${this.calEvent.language("additional_notes")}:</strong><br />${this.getDescriptionText()}<br />` +
-      this.getChangeEventFooterHtml()
-    );
-  }
-
-  /**
-   * Conditionally returns the event's location. When VideoCallData is set,
-   * it returns the meeting url. Otherwise, the regular location is returned.
-   * For Daily video calls returns the direct link
-   * @protected
-   */
-  protected getLocation(): string | null | undefined {
-    const isDaily = this.calEvent.location === "integrations:daily";
-    if (this.calEvent.videoCallData) {
-      return this.calEvent.videoCallData.url;
-    }
-    if (isDaily) {
-      return process.env.BASE_URL + "/call/" + this.getUid();
-    }
-    return this.calEvent.location;
-  }
-
-  /**
-   * Returns the event's description text. If VideoCallData is set, it prepends
-   * some video call information before the text as well.
-   *
-   * @protected
-   */
-  protected getDescriptionText(): string | null | undefined {
-    if (this.calEvent.videoCallData) {
-      return `
-${this.calEvent.language("integration_meeting_id", {
-  integrationName: getIntegrationName(this.calEvent.videoCallData.type),
-  meetingId: this.calEvent.videoCallData.id,
-})}
-${this.calEvent.language("password")}: ${this.calEvent.videoCallData.password}
-${this.calEvent.description}`;
-    }
-    return this.calEvent.description;
-  }
-
-  /**
-   * Returns an extended description with all important information (as plain text).
-   *
-   * @protected
-   */
-  public getRichDescription(): string {
-    return stripHtml(this.getRichDescriptionHtml());
-  }
-
-  /**
-   * Returns a calendar event with rich description.
-   */
-  public asRichEvent(): CalendarEvent {
-    const eventCopy: CalendarEvent = { ...this.calEvent };
-    eventCopy.description = this.getRichDescriptionHtml();
-    eventCopy.location = this.getLocation();
-    return eventCopy;
-  }
+export const getUid = (calEvent: CalendarEvent) => {
+  return calEvent.uid ?? translator.fromUUID(uuidv5(JSON.stringify(calEvent), uuidv5.URL));
+};
 
-  /**
-   * Returns a calendar event with rich description as plain text.
-   */
-  public asRichEventPlain(): CalendarEvent {
-    const eventCopy: CalendarEvent = { ...this.calEvent };
-    eventCopy.description = this.getRichDescription();
-    eventCopy.location = this.getLocation();
-    return eventCopy;
-  }
-}
+export const getCancelLink = (calEvent: CalendarEvent) => {
+  return process.env.BASE_URL + "/cancel/" + getUid(calEvent);
```

Comment:
Can we import and use `BASE_URL` from `lib/config/constants.ts` here as well?

Reply:
Yes @alishaz-polymath I was just testing if it works on vercel deployments and it seems like it is.